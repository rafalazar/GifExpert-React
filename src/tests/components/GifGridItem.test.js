import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Soy un título';
    const url = 'https://localhost/something.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe tener párrafo con el title', () => {

        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
        
    });

    test('Debe tener src y alt igual que los props', () => {
        const img = wrapper.find('img');
        const {src, alt} = img.props();
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });
    
    test('Debe tener la clase animate__fadeIn', () => {
        const div = wrapper.find('div');
        const { className } = div.props();
        
        expect(className.includes('animate__fadeIn')).toBe(true);
    });
    
});
