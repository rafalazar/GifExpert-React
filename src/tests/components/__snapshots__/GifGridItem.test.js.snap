// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrat el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
